pkg/backend/filestate/backend.go

```diff
@@ -233,19 +237,117 @@ func New(ctx context.Context, d diag.Sink, originalURL string, project *workspac
 	}
 	backend.currentProject.Store(project)
 
+	// projectMode tracks whether the current state supports project-scoped stacks.
```

Comment:
I think most of these changes should be part of https://github.com/pulumi/pulumi/pull/12437

Especially the warning part

Reply:
👍 perfect

Reply:
Moved to other PR.